frontend/public/js/util.js

```diff
@@ -10,4 +10,31 @@
       function () { return $(this).data(prop) == val; }
     );
   };
-})(window.jQuery);
\ No newline at end of file
+})(window.jQuery);
+
+
+JSON.flatten = function(data) {
+    var result = {};
+    function recurse (cur, prop) {
+        if (Object(cur) !== cur) {
+            result[prop] = cur;
+        } else if (Array.isArray(cur)) {
+          /*
+             for(var i=0, l=cur.length; i<l; i++)
```

Comment:
hmm not sure.
so right now the flattening works with nested objects

for example,
{
    config: {
        l1: 20
    }
}

will get flattened to 
{
    config.l1: 20
}

The commented code was for lists inside an object, an it would just use the index when flattening. Wasn't sure how to support flattening lists properly, since we probably don't want the order to matter. So I commented it out for now, and will revisit how to deal lists in the json later.